core/services/job/spawner.go

```diff
@@ -176,55 +186,74 @@ func (js *spawner) startUnclaimedServices() {
 	js.jobTypeDelegatesMu.RLock()
 	defer js.jobTypeDelegatesMu.RUnlock()
 
-	for _, job := range jobs {
-		if _, exists := js.services[job.ID]; exists {
-			logger.Warnw("Job spawner ORM attempted to claim locally-claimed job, skipping", "jobID", job.ID)
+	js.activeJobsMu.Lock()
```

Comment:
Using two mutexes here seems like it could more likely end us up in a deadlock situation, could we simplify this to one?

Reply:
`jobsTypeDelegates` seems to be immutable, do we need a lock for it at all?